### ISSUE TITLE
Implement "Generate constructor" (the `new` method)

### ISSUE THREAD

**Issue by dvtomas** [intellij-rust/intellij-rust#4937]
Oftentimes I write code like this:

```rust
struct S {
  foo: Foo,
  bar: Bar,
  baz: Baz
}

impl S {
  pub fn new(foo: Foo, bar: Bar, baz: Baz) -> Self {
    S { foo, bar, baz }
  }
}
```

It would be nice if we could have the `new` method auto-generated, e.g. by using `Alt-Insert` when on the struct or inside the impl block.

**Comment by Undin** [comment]
@dvtomas thanks for the suggestion!
Do you know that almost the same feature is already implemented in #2815?
Just invoke `Code | Generate` action and choose `Generate Constructor` option. Note, it works only when you caret inside the corresponding struct.
![2020-02-09 02 00 15](https://user-images.githubusercontent.com/2539310/74093197-eea25580-4adf-11ea-8a3d-4bd62e0b506b.gif)

And, indeed, it would be great if the plugin provides this action inside the corresponding `impl` block

**Comment by dvtomas** [comment]
Aha, I didn't know about that. That's exactly what I had on mind. I guess the problem was i was executing the "Generate" pop-up inside the `impl` block, not on the struct itself, and inside the `impl` block it doesn't offer this action. Maybe it would be nice to offer it in the `impl` context as well.

Anyway, thanks!
